src/application/tally/widgets/proposal.tsx

```diff
@@ -0,0 +1,136 @@
+        className,
+      )}
+      top={top}
+      onClose={onClose}
+    >
+      <header className="mb-auto flex items-center justify-between space-x-3">
+        <p className="line-clamp-[1] break-all text-xs text-tally-gray-600">
+          By{' '}
+          <a
+            href={
+              proposalDetails.creator.ens.length > 0 ||
+              proposalDetails.creator.address.length > 0
+                ? getOrganizationDelegateUrl(
+                    proposalDetails.organization.slug ?? '',
+                    proposalDetails.creator.ens.length > 0
+                      ? proposalDetails.creator.ens
+                      : proposalDetails.creator.address,
+                  )
+                : getOrganizationUrl(proposalDetails.organization.slug ?? '')
+            }
+            className="font-semibold font-semibold hover:underline"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            {proposalDetails.creator.name.length > 0
+              ? proposalDetails.creator.name
+              : proposalDetails.creator.address}
+          </a>
+        </p>
+        <Chip
+          className={classes(
+            'rounded-sm border-tally-border-primary px-[4px] py-0.5 font-bold uppercase tracking-wide',
+            getStatusBadgeColorClassNames(proposalDetails.status),
+          )}
+        >
+          {proposalDetails.status}
+        </Chip>
+      </header>
+      <main className="my-2 grid">
+        <p className="line-clamp-[1] break-all text-base font-bold tracking-tighter">
+          {proposalDetails.metadata.title}
+        </p>
+        <p className="mt-1 line-clamp-[4] overflow-hidden">
+          {proposalDetails.metadata.description}
+        </p>
+      </main>
+      <footer className="mt-auto flex items-center justify-between">
+        <div className="flex justify-start gap-1.5">
+          <div className="flex min-w-[84px] items-center text-xs font-semibold font-semibold leading-5 text-tally-gray-500">
+            {getEndsInLabel(getDifferenceInDays(proposalEndDateInMs))}
+          </div>
+          <a
+            href={getProposalUrl(
+              proposalDetails.organization.slug,
+              proposalDetails.id,
+            )}
+            rel="noopener noreferrer"
+            target="_blank"
+          >
+            <Chip
+              className=" mr-2 inline-flex h-7 select-none items-center justify-center rounded-md border border-tally-gray-700 bg-tally-gray-700 px-4 py-2 text-sm font-semibold leading-6 text-white transition-all duration-200 ease-in-out hover:bg-tally-gray-900 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50"
+              variant="info"
+              width="long"
+            >
+              {proposalDetails.status.toLowerCase() === 'active'
+                ? 'Vote'
+                : 'View'}
+            </Chip>
```

Comment:
do we know finite list of proposal `status`? we could maybe narrow the type and make a constant to map status to label and apply fn on top of it:
```ts
const STATUS_TO_LABEL: Record<Status, string> = {
  active: 'active',
  foo: 'view',  
}

const getStatusLabel = (status: Status) => {
  return STATUS_TO_LABEL[status];
}
```

Reply:
I posted statuses image above and also mentioned it's fine to have status as string so here we could just make a fn to return active if status is active or view if other